samples/openapi3/client/petstore/python/petstore_api/configuration.py

```diff
@@ -169,6 +172,31 @@ def __init__(self, host="http://petstore.swagger.io:80/v2",
         # Disable client side validation
         self.client_side_validation = True
 
+    @classmethod
+    def set_default(cls, default):
+        """Set default instance of configuration.
+
+        It stores default configuration, which can be
+        returned by get_instance method.
+
+        :param default: object of Configuration
+        """
+        cls._default = copy.copy(default)
+
+    @classmethod
+    def get_instance(cls):
+        """Return new instance of configuration.
+
+        This method returns newly created, based on default constructor,
+        object of Configuration class or returns a copy of default
+        configuration passed by the set_default method.
+
+        :return: The configuration object.
+        """
+        if cls._default is not None:
+            return copy.copy(cls._default)
```

Comment:
Why copy it vs just return the instance?
The CI tests are failing when trying to deepcopy the logger.

Reply:
How about returning a new Configuration instance which uses the same instantiation inputs as the default instance?

Reply:
Yes, deepcopy doesn't work. Here as in the previous implementation a copy of default configuration is returned.

The CI tests failed because a remote endpoint returned 404 error on add_pet method. I don't know why, I'll try to run it again.

Reply:
Tests passed. I guess the original intention was to return a copy which looks like a new object but with different default values. Maybe I should name the method `new_instance` instead of `get_instance` to make it more understandable... ?

Reply:
I changed method name to `new_instance()`.

Reply:
new_instance is good.
How do you feel about get_default_copy as a name?

Reply:
Updated, both versions look good to me.